tests/monitoring/component_monitoring.go

```diff
@@ -0,0 +1,348 @@
+		restErrorsBurtsAlert: "VirtApiRESTErrorsBurst",
+		restErrorsHighAlert:  "VirtApiRESTErrorsHigh",
+		lowCountAlert:        "LowVirtAPICount",
+	}
+	virtController = alerts{
+		deploymentName:       "virt-controller",
+		downAlert:            "VirtControllerDown",
+		noReadyAlert:         "NoReadyVirtController",
+		restErrorsBurtsAlert: "VirtControllerRESTErrorsBurst",
+		restErrorsHighAlert:  "VirtControllerRESTErrorsHigh",
+		lowCountAlert:        "LowVirtControllersCount",
+	}
+	virtHandler = alerts{
+		deploymentName:       "virt-handler",
+		restErrorsBurtsAlert: "VirtHandlerRESTErrorsBurst",
+		restErrorsHighAlert:  "VirtHandlerRESTErrorsHigh",
+	}
+	virtOperator = alerts{
+		deploymentName:       "virt-operator",
+		downAlert:            "VirtOperatorDown",
+		noReadyAlert:         "NoReadyVirtOperator",
+		restErrorsBurtsAlert: "VirtOperatorRESTErrorsBurst",
+		restErrorsHighAlert:  "VirtOperatorRESTErrorsHigh",
+		lowCountAlert:        "LowVirtOperatorCount",
+	}
+)
+
+var _ = Describe("[Serial][sig-monitoring]Component Monitoring", Serial, decorators.SigMonitoring, func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+	var scales *Scaling
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+	})
+
+	increaseRateLimit := func() {
+		rateLimitConfig := &v1.ReloadableComponentConfiguration{
+			RestClient: &v1.RESTClientConfiguration{
+				RateLimiter: &v1.RateLimiter{
+					TokenBucketRateLimiter: &v1.TokenBucketRateLimiter{
+						Burst: 300,
+						QPS:   300,
+					},
+				},
+			},
+		}
+		originalKubeVirt := util.GetCurrentKv(virtClient)
+		originalKubeVirt.Spec.Configuration.ControllerConfiguration = rateLimitConfig
+		originalKubeVirt.Spec.Configuration.HandlerConfiguration = rateLimitConfig
+		tests.UpdateKubeVirtConfigValueAndWait(originalKubeVirt.Spec.Configuration)
+	}
+
+	reduceAlertPendingTime := func() {
+		By("Reducing alert pending time")
+		newRules := getPrometheusAlerts(virtClient)
+		var re = regexp.MustCompile("\\[\\d+m\\]")
+
+		var gs []promv1.RuleGroup
+		for _, group := range newRules.Spec.Groups {
+			var rs []promv1.Rule
+			for _, rule := range group.Rules {
+				var r promv1.Rule
+				rule.DeepCopyInto(&r)
+				if r.Alert != "" {
+					r.For = "0m"
+					r.Expr = intstr.FromString(re.ReplaceAllString(r.Expr.String(), `[1m]`))
+					r.Expr = intstr.FromString(strings.ReplaceAll(r.Expr.String(), ">= 300", ">= 0"))
+				}
+				rs = append(rs, r)
+			}
+
+			gs = append(gs, promv1.RuleGroup{
+				Name:  group.Name,
+				Rules: rs,
+			})
+		}
+		newRules.Spec.Groups = gs
+
+		updatePromRules(virtClient, &newRules)
+	}
+
+	Context("Up metrics", func() {
+		BeforeEach(func() {
+			scales = NewScaling(virtClient, []string{virtOperator.deploymentName, virtController.deploymentName, virtApi.deploymentName})
+			scales.UpdateScale(virtOperator.deploymentName, int32(0))
+
+			reduceAlertPendingTime()
+		})
+
+		AfterEach(func() {
+			scales.RestoreAllScales()
+
+			time.Sleep(10 * time.Second)
```

Comment:
Is this mandatory? I see several other usages of these sleep. Can you explain why since later we use an Eventually loop?

Reply:
the restoration will always take some time, my idea here was to reduce the amount of computation done in eventually, but this is totally removable

Reply:
That's ok for me.
Anyway, if you want to reduce the computation you can also increase the interval in the `Eventually` block :)